server/lib/pbench/test/unit/test_process_tb.py

```diff
@@ -16,183 +16,185 @@ def get(self, section: str, option: str):
-            return [tb_path]
-
         def mock_get(self, section: str, option: str):
             if option == "pbench-receive-dir-prefix":
                 return tb_path
             if option == "put-token":
                 return "Authorization-token"
-            assert False
+            assert (
+                False
+            ), f"Unexpected configuration option, {section!r}/{option!r}, in mocked config.get()"
 
         def mock_results_push(self, ctrl, tb, token):
             raise RuntimeError(expected_error_msg)
 
         expected_result = Results(nstatus="", ntotal=1, ntbs=0)
-        monkeypatch.setattr(Path, "glob", mock_glob)
         monkeypatch.setattr(MockConfig, "get", mock_get)
         monkeypatch.setattr(ProcessTb, "results_push", mock_results_push)
-        ptb = ProcessTb(MockConfig(), self.logger)
+        ptb = ProcessTb(MockConfig(), test_logger)
         res = ptb.process_tb()
 
         assert res == expected_result
         assert expected_error_msg in caplog.text
 
-    def test_process_tb(self, monkeypatch):
+    def test_process_tb(self, monkeypatch, test_logger):
         """verify processing of tar balls without any failure"""
         tb_path = fixture_path / "tarball"
+        good_tb = tb_path / tarball
 
-        def mock_remove(*args):
-            pass
-
-        def mock_glob(*args, **kwargs):
-            tb_path = fixture_path / "tarball-002" / "log.tar.xz.md5"
-            return [tb_path]
+        def mock_remove(path, *, dir_fd=None):
+            if path == good_tb:
+                pass
+            assert False, "Unexpected tar ball in mocked os.remove()"
 
         def mock_get(self, section: str, option: str):
             if option == "pbench-receive-dir-prefix":
                 return tb_path
             if option == "put-token":
                 return "Authorization-token"
-            assert False
+            assert (
+                False
+            ), f"Unexpected configuration option, {section!r}/{option!r}, in mocked config.get()"
 
         def mock_results_push(self, ctrl, tb, token):
             return
 
         expected_result = Results(nstatus=f": processed {tarball}\n", ntotal=1, ntbs=1)
         monkeypatch.setattr(os, "remove", mock_remove)
-        monkeypatch.setattr(Path, "glob", mock_glob)
         monkeypatch.setattr(MockConfig, "get", mock_get)
         monkeypatch.setattr(ProcessTb, "results_push", mock_results_push)
-        ptb = ProcessTb(MockConfig(), self.logger)
+        ptb = ProcessTb(MockConfig(), test_logger)
         res = ptb.process_tb()
 
         assert res == expected_result
 
-    def test_process_tb_zero(self, monkeypatch):
+    def test_process_tb_zero(self, monkeypatch, test_logger):
         """verify processing if there are no TBs without any failure"""
         tb_path = fixture_path / "tarball"
 
-        def mock_glob(*args, **kwargs):
+        def mock_glob(self, pattern):
             return []
```

Comment:
So, I think `self` refers to a `Path`, so you can assert that `self == tb_path`, and you can assert that `pattern == "**/*.tar.xz.md5"`.

Reply:
Sorry, I didn't understand the requirement of assert here